### PR TITLE
Fix #698 and #699

### LINKED ISSUE #698: help does not work on tables
The builtin `help` function does not seem to work on tables.
`help(MyTable)` raises `DataJointError: Class UserTable is not properly declared (schema decorator not applied?)`, all other table functionality being intact.

### LINKED ISSUE #699: Append table definition to docstring
Thinking about improving the documentation of our own database, we thought it would be nice to add the table definition to its docstring. We do this using a decorator on the table definition somewhat like this:
```python
def docappend(cls):
    cls.__doc__ += cls.definition
    return cls

@docappend
@schema
class MyTable(dj.Imported):
    .....
```
To us, it seemed like a minor but useful feature to include in a future version of datajoint.

### PR DESCRIPTION
**Pull request by dimitri-yatsenko**
* Fix #698 -- Help now works for DataJoint classes
* Fix #699 -- table definition is appended to doc string

### CODE REVIEW

File: tests/test_relation.py [hunk 1]
```diff
@@ -36,6 +36,12 @@ def setup_class(cls):
         cls.img = schema.Image()
         cls.trash = schema.UberTrash()
 
+    def test_class_help(self):
+        help(schema.TTest)
```
Comment (ixcat) [inline tests/test_relation.py]: would it make sense to also e.g.:
`assert schema.TTest.definition in schema.TTest.__doc__`

File: tests/test_relation.py [hunk 2]
```diff
@@ -36,6 +36,12 @@ def setup_class(cls):
...
+
+    def test_instance_help(self):
```
Comment (ixcat) [inline tests/test_relation.py]: as before r.e. testing for definition in doc.

File: tests/test_relation.py (hunk 1 above)
Comment (dimitri-yatsenko) [reply tests/test_relation.py]: done. Also moved into `test_declare`

File: tests/test_relation.py (hunk 2 above)
Comment (dimitri-yatsenko) [reply tests/test_relation.py]: done. Also moved into `test_declare`

File: datajoint/table.py [hunk 3]
```diff
@@ -45,14 +45,9 @@ def heading(self):
         """
         if self._heading is None:
             self._heading = Heading()  # instance-level heading
-        if not self._heading:  # lazy loading of heading
-            if self.connection is None:
-                raise DataJointError(
-                    'DataJoint class is missing a database connection. '
-                    'Missing schema decorator on the class? (e.g. @schema)')
```
[the remark this answers is not in the document]
Comment (dimitri-yatsenko) [reply datajoint/table.py]: If the table is not decorated, with a `@schema`, then another error is already raised in `user_tables.py` when attempting to access the `database` property. Here we must be able to return a `None` connection object for the help to be properly generated. Help actually instantiates parent classes and this was one of the reasons help would raise an error.

**Comment by guzman-raphael** [comment]
@dimitri-yatsenko Can you also update `docs-parts/intro/Releases_lang1.rst` to match `CHANGELOG.md`?

File: datajoint/blob.py [hunk 4]
```diff
@@ -253,6 +253,9 @@ def pack_recarray(self, array):
     def read_sparse_array(self):
         raise DataJointError('datajoint-python does not yet support sparse arrays. Issue (#590)')
 
+    def read_scalar(selfs):
+
```
Comment (guzman-raphael) [inline datajoint/blob.py]: This looks like a typo... Does not raise/return anything?

**Pull request merged**